lightning-invoice/src/payment.rs

```diff
@@ -269,6 +269,10 @@ where
 						final_value_msat: invoice.amount_milli_satoshis().or(amount_msats).unwrap(),
 						final_cltv_expiry_delta: invoice.min_final_cltv_expiry() as u32,
 					};
+					if has_expired(&params) {
+						log_trace!(self.logger, "Invoice expired prior to first send for payment {}", log_bytes!(payment_hash.0));
+						return Err(PaymentError::Invoice("Invoice expired prior to send"));
+					}
```

Comment:
You can do this prior to creating `RouteParameters` by calling `invoice.is_expired()`.